### PR TITLE
Blog App: validations and model specs

### PR DESCRIPTION
**Pull request by dheerajsachdeva**
- Added the following validations:
- For the User model: Name must not be blank.
- PostsCounter must be an integer greater than or equal to zero.
- For the Post model: Title must not be blank.
- Title must not exceed 250 characters.
- CommentsCounter must be an integer greater than or equal to zero.
- LikesCounter must be an integer greater than or equal to zero.
- Added unit specs for all of models' methods and validations.

All tests are passing: 

image

😄

### CODE REVIEW

File: spec/post_spec.rb [hunk 1]
```diff
@@ -0,0 +1,34 @@
+require 'rails_helper'
+
+RSpec.describe Post, type: :model do
+  user = User.create(name: 'Tom', photo: 'https://unsplash.com/photos', bio: 'Teacher from Mexico.')
+  post = Post.new(author: user, title: 'post1', text: 'This is my first Post')
+
+  before { post.save }
+
```
Comment (Danieldotcomcoder) [inline spec/post_spec.rb]: Kindly make sure to add test for the validation 'Title must not exceed 250 characters.'  and also for Comments and Likes counters of the post.

**Review by Danieldotcomcoder — CHANGES_REQUESTED** [review]
Hello @dheerajsachdeva , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights
- Professional Readme File ✔✔
- Linters are setup correctly ✔✔
- Clear and nice code ✔✔

## Required Changes ♻️

Some tests are failing , kindly make sure all tests are passing:
image


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)

File: spec/post_spec.rb (hunk 1 above)
Comment (dheerajsachdeva) [reply spec/post_spec.rb]: @Danieldotcomcoder Can you please explain more which test is failing

I am getting all tests passed. See below:

image

Comment (dheerajsachdeva) [reply spec/post_spec.rb]: @Danieldotcomcoder Test for the validation 'Title must not exceed 250 characters.' and also for Comments and Likes counters of the post added

**Review by franclobo — APPROVED** [review]
Hi @dheerajsachdeva, 

# STATUS: APPROVED 😃 

git-merge

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- No linter errors ✔️ 
- Descriptive PR ✔️ 
- Good communication with CR ✔️ 
- All tests complete ✔️ 
- All tests passed ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**